manticore/core/state.py

```diff
@@ -176,7 +176,8 @@ def new_symbolic_buffer(self, nbytes, **options):
         :return: :class:`~manticore.core.smtlib.expression.Expression` representing the buffer.
         '''
         name = options.get('name', 'buffer')
-        expr = self.constraints.new_array(name=name, index_max=nbytes)
+        taint = options.get('taint', frozenset())
```

Comment:
Let's also update the docstrings for new_symbolic_buffer and symbolicate_buffer. adding something like this

```
        :param taint: Taint identifier of this buffer
        :type taint: tuple or frozenset
```